R/utils-numeric.R

```diff
@@ -1,3 +1,66 @@
+## TODO think about the relationship between this and the plot.data bin fxns
+#' Find Bin Ranges
+#' 
+#' This function will find bin start, end and labels for a 
+#' continuous variable. Optionally, it can return a value/ count
+#' per bin. By default returns 10 bins for equalRanges and quantile
+#' methods and 6 for sd (standard deviations).
+#' @param x Numeric (or Date) vector to find bins for
+#' @param method A string indicating which method to use to find bins ('equalRanges', 'quantile', 'sd')
+#' @param numBins A number indicating how many bins are desired
+#' @param getValue A boolean indicating whether to return the counts per bin
+#' @export 
+getBinRanges <- function(x, method = c('equalInterval', 'quantile', 'sd'), numBins = NULL, getValue = c(TRUE, FALSE)) {
+  method <- veupathUtils::matchArg(method)
+  getValue <- veupathUtils::matchArg(getValue)
+  if (is.null(numBins)) numBins <- 10
+
+  isDate <- FALSE
+  if (class(x) == 'Date') {
+    x <- as.numeric(x)
+    isDate <- TRUE
+  }
+
+  binEdges <- unname(breaks(x, method, numBins))
+  if (anyDuplicated(binEdges)) {
+    warning("There is insufficient data to produce the requested number of bins. Returning as many bins as possible.")
+    binEdges <- unique(binEdges)
+  }
+
+  binStart <- binEdges[1:(length(binEdges)-1)]
```

Comment:
I was confused here until I figured out they were zipped together later. Could it be named `binStarts` ?